helper/augroups.nix

```diff
@@ -0,0 +1,189 @@
+          Can return true to delete the autocommand, and receives a table argument (opts) with these keys:
+             • id: (number) autocommand id
+             • event: (string) name of the triggered event |autocmd-events|
+             • group: (number|nil) autocommand group id, if any
+             • match: (string) expanded value of |<amatch>|
+             • buf: (number) expanded value of |<abuf>|
+             • file: (string) expanded value of |<afile>|
+             • data: (any) arbitrary data passed from |nvim_exec_autocmds()|
+
+          Will be expanded to:
+
+          ```lua
+          function(opts)
+            ''${luaCallback}
+          end
+          ```
+        '';
+        example = ''
+          vim.highlight.on_yank {
+            higroup = (
+              vim.fn['hlexists'] 'HighlightedyankRegion' > 0 and 'HighlightedyankRegion' or 'IncSearch'
+            ),
+            timeout = 200,
+          }
+        '';
+      };
+      vimCallback = mkOption {
+        default = null;
+        type = nullOr str;
+        description = ''
+          Vimscript function name called when the event(s) is triggered.
+
+          Conflicts with luaCallback
+        '';
+      };
+      command = mkOption {
+        default = null;
+        type = nullOr str;
+        description = ''
+          Vim command to execute on event. Cannot be used with {lua,vim}Callback
+        '';
+      };
+      once = mkOption {
+        default = false;
+        type = bool;
+        description = ''
+          Run the autocommand only once |autocmd-once|
+        '';
+      };
+      nested = mkOption {
+        default = false;
+        type = bool;
+        description = ''
+          Run nested autocommands |autocmd-nested|.
+        '';
+      };
+    };
+  };
+
+  quote = string: "\"${string}\"";
+
+  toLuaStringList = strings:"{${concatMapStringsSep "," quote strings}}";
+
+  genAutocmd = group: { event
+               , pattern
+               , buffer
+               , desc
+               , luaCallback
+               , vimCallback
+               , command
+               , once
+               , nested
+               }: let
+               # TODO: Assert that only one of luaCallback vimCallback or command is set
+               events = if builtins.isList event then toLuaStringList event else "{${quote event}}";
+               opts = {
+                 inherit group pattern buffer desc command once nested;
+                 callback = if luaCallback == null then
+                   vimCallback
+                 else
+                   rawLua ''
+                   function(opts)
+                     ${luaCallback}
+                   end
+                   '';
+               };
+             in ''
+    do
+      local events = ${events}
+      local opts = ${toLuaObject opts}
+      vim.api.nvim_create_autocmd(events, opts)
+    end
+  '';
+
+  genAugroup = {name, autocmds, clear}: let
```

Comment:
Could you add brief documentations for this function?

Also for genAugroups and genAutocmd